File: ido_sdk_flutter/ido_sdk/bluetooth_sdk/android/src/main/kotlin/com/example/flutter_bluetooth/ble/callback/ScanCallBack.java
package com.example.flutter_bluetooth.ble.callback;

import com.example.flutter_bluetooth.dfu.BLEDevice;

/**
 * 扫描手环设备监听回调
 * <br/>
 * The callback for scan device.
 */
public interface ScanCallBack {


    /**
     * 扫描开始
     * </>
     * Scan start.
     */
    void onStart();

    /**
     * 扫到一个手环设备
     * <br/>
     * Find a device.
     */
    void onFindDevice(BLEDevice device);

    /**
     * 扫描结束
     * <br/>
     * Scan finished.
     */
    void onScanFinished();

}

File: ido_sdk_flutter/ido_sdk/bluetooth_sdk/android/src/main/kotlin/com/example/flutter_bluetooth/ble/scan/ScanCallBack.kt
package com.example.flutter_bluetooth.ble.scan

/**
 * 扫描手环设备监听回调
 * <br></br>
 * The callback for scan device.
 */
interface ScanLeCallBack {

    /**
     * 扫描开始
     * >
     * Scan start.
     */
    fun onScanLeStart()

    /**
     * 扫到一个手环设备
     * <br></br>
     * Find a device.
     */
    fun onScanLeResult(scanResult: Map<String, Any?>)

    /**
     * 扫描结束
     * <br></br>
     * Scan finished.
     */
    fun onScanLeFinished()
}